File: day13/2.py
from itertools import count

with open("input-test.txt") as f:
    inp = [line.strip() for line in f.readlines()][1]
    inp = inp.split(",")

nums = [(i, int(n)) for i, n in enumerate(inp) if n != "x"]

ans = nums[0][1]
step = 1

for i, n in nums:

    while True:
        if (ans + i) % n == 0:
            ans = ans + step
            # ans = next(c for c in count(ans, step) if (c + i) % n == 0)
            break
        step *= n

print(ans)


next(count(100, 10))

File: day18/2.jl
import Base: -,/

struct D <: Number
    val::Int64
end

-(x::D, y::D) = D(x.val * y.val) 
/(x::D, y::D) = D(x.val + y.val) 

f = open("day18/input.txt")
inp = readlines(f)
close(f)

inp = [replace(line, "*" => "-") for line in inp]
inp = [replace(line, "+" => "/") for line in inp]

function conv(line)
    num_inds = findall(r"\d", line)
    nums = Set([line[index] for index in num_inds])
    for n in nums
        line = replace(line, n => string("D(", n, ")"))
    end
    return line
end

s = 0
for line in inp
    sym = Meta.parse(conv(line))
    s += eval(sym).val
end
print(s)

